extensions/podman/src/compatibility-mode.ts

```diff
@@ -0,0 +1,112 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import * as extensionApi from '@podman-desktop/api';
+import * as sudo from 'sudo-prompt';
+import * as fs from 'fs';
+
+// Create an abstract class for compatibility mode (macOS only)
+// TODO: Windows, Linux
+abstract class SocketCompatibility {
+  abstract enable(): Promise<void>;
+  abstract disable(): Promise<void>;
+  abstract details: string;
+}
+
+export class DarwinSocketCompatibility extends SocketCompatibility {
+  // Shows the details of the compatibility mode on what we do.
+  details = `The podman-mac-helper binary will be ran in order to enable or disable compatibility mode. 
+    This requires administrative privileges.`;
+
+  findPodmanHelper(): string {
+    const homebrewPath = '/opt/homebrew/bin/podman-mac-helper';
+    const podmanPath = '/opt/podman/bin/podman-mac-helper';
+
+    if (fs.existsSync(homebrewPath)) {
+      return homebrewPath;
+    } else if (fs.existsSync(podmanPath)) {
+      return podmanPath;
+    } else {
+      return '';
+    }
+  }
+  // Run sudo command for podman mac helper
+  async runSudoMacHelperCommand(command: string): Promise<void> {
+    const sudoOptions = {
+      name: 'Podman Desktop Compatibility Mode',
+    };
+    return new Promise((resolve, reject) => {
+      sudo.exec(command, sudoOptions, (error, stdout, stderr) => {
+        console.log('sudo command output: ', stdout, stderr);
+
+        // podman-mac-helper does not error out on failure for some reason, so we need to check the output for
+        // 'Error:' to determine if the command failed despite the exit code being 0
+        // Issue: https://github.com/containers/podman/issues/17785
+        // we'll most likely need to keep this check for old releases of podman-mac-helper.
+        if (stderr?.includes('Error:')) {
+          reject(stderr);
+        }
+
+        if (error) {
+          reject(error);
+        } else {
+          resolve();
+        }
+      });
+    });
+  }
+
+  async runCommand(command: string, description: string): Promise<void> {
+    // Find the podman-mac-helper binary
+    const podmanHelperBinary = this.findPodmanHelper();
+    if (podmanHelperBinary === '') {
+      extensionApi.window.showErrorMessage('podman-mac-helper binary not found.', 'OK');
+      return;
+    }
+
+    const fullCommand = `${podmanHelperBinary} ${command}`;
+    try {
+      await this.runSudoMacHelperCommand(fullCommand);
+      extensionApi.window.showInformationMessage(`Docker socket compatibility mode for Podman has been ${description}.
+      Restart your Podman machine to apply the changes.`);
```

Comment:
I might expect to have a message prompting me: click to Restart now/Later and the extension restart for me the machine for me (could be a separate/later improvement) but I think UX would be better